### PR TITLE
feat: [IAC-826]: Update components passed to MFE for k8s advanced config

### PR DESCRIPTION
**Pull request by bgribben-harness**
- Updates components passed from core-ui to the IACM MFE for use in K8s advanced stage config.
- Updates `StageInputSetForm` tests to include a basic test for rendering IACM Form branch

### CODE REVIEW

**Review by Harness-JKhattak — DISMISSED** [review]
🥦

File: src/modules/85-iacm/interfaces/IACMCustomMicroFrontendProps.types.ts [hunk 1]
```diff
@@ -57,5 +60,8 @@ export interface IACMCustomMicroFrontendProps {
     MultiTypeSecretInput: typeof MultiTypeSecretInput
     Duration: typeof Duration
     DefaultConsoleViewStepDetails: typeof DefaultConsoleViewStepDetails
+    FormMultiTypeCheckbox: typeof FormMultiTypeCheckbox
+    ConfigureOptions: typeof ConfigureOptions
+    FormMultiTypeDuration: typeof FormMultiTypeDuration
```
Comment (knagurski) [inline src/modules/85-iacm/interfaces/IACMCustomMicroFrontendProps.types.ts:65]: You'll probably need to bump the `microfrontends` package version to get these into the types for your MFE to consume

File: src/modules/70-pipeline/components/PipelineInputSetForm/__tests__/StageInputSetForm.test.tsx [hunk 2]
```diff
@@ -412,5 +417,33 @@ describe('stageinputset tests', () => {
...
+      const { getByText } = render(
+        <TestWrapper defaultFeatureFlagValues={{ IACM_ENABLED: true }}>
+          <Formik formName="test-form" initialValues={{}} onSubmit={jest.fn()}>
+            <Form>
+              <StageInputSetForm {...IACMprops} viewType={StepViewType.DeploymentForm} />
+            </Form>
+          </Formik>
+        </TestWrapper>
+      )
+      expect(getByText('IACMInputSet')).toBeTruthy()
```
Comment (knagurski) [inline src/modules/70-pipeline/components/PipelineInputSetForm/__tests__/StageInputSetForm.test.tsx:446]: Makes zero difference to the result of your test, but `toBeInTheDocument()` conveys what you're trying to prove better than `toBeTruthy()`
```suggestion
      expect(getByText('IACMInputSet')).toBeInTheDocument()
```

**Review by udayvunnam — DISMISSED** [review]

File: src/modules/70-pipeline/components/PipelineInputSetForm/__tests__/mocks.tsx [hunk 3]
```diff
@@ -687,3 +689,11 @@ export const getCICodebaseParallelTemplateProps = ({ formik }: { formik: any }):
     orgIdentifier: 'default'
   }
 })
+
+export function MockedIACMMFE({ children }: { children: ReactElement }): ReactElement {
```
Comment (knagurski) [inline src/modules/70-pipeline/components/PipelineInputSetForm/__tests__/mocks.tsx:693]: Nothing really wrong here, but `PropsWithChildren` is a great type to use for things like this.
```suggestion
export function MockedIACMMFE({ children }: PropsWithChildren<unknown>): ReactElement {
```

Even better when you can use an arrow function and `FC` (some modules don't allow arrow functions for some really silly reason).
```suggestion
export const MockedIACMMFE: FC<PropsWithChildren<unknown>> = ({ children }) => {
```

Comment (bgribben-harness) [reply src/modules/70-pipeline/components/PipelineInputSetForm/__tests__/mocks.tsx:693]: TIL!

Comment (knagurski) [reply src/modules/70-pipeline/components/PipelineInputSetForm/__tests__/mocks.tsx]: If you're wondering about the `unknown` bit, it just means that we're only specifying the `children` prop. If you had other props, you can put them in there instead.

```
interface MyComponentProps {
  prop1: string
  prop2: boolean
}

const MyComponent:FC<PropsWithChildren<MyComponentProps>> = ({ children, prop1, prop2 }) => {
  // ...
}
```

**Review by knagurski — APPROVED** [review]
Noice

**Review by Harness-JKhattak — APPROVED** [review]

**Comment by bgribben-harness** [comment]
retrigger cypress-pipeline

**Comment by bgribben-harness** [comment]
retrigger cypress-pipeline

**Review by udayvunnam — APPROVED** [review]

**Pull request merged**